src/utils.js

```diff
@@ -44,34 +44,45 @@ export default class Utils {
         }));
     }
 
-    qsM(s, t, d = null) {
+    qsM(s, t, d = false) {
         let e = null; // Trackable Element
         let c = 'button';
         let p = []; // Path
         if (t.nodeType === 3) {
             t = t.parentNode;
         }
         while (t && t !== window.parent.document) {
-            let matches = (t.matches || t.msMatchesSelector || function () {
-                return false;
-            }).bind(t);
-            if (!d) {
-                if (matches(s)) {
-                    return {
-                        'element': t
-                    };
+            let matches = (
+                t.matches ||
+                t.msMatchesSelector ||
+                function () {
+                    return false;
                 }
-            } else {
+            ).bind(t);
+
+            if (d) {
                 if (t.hasAttribute(d)) {
                     p.unshift(t.getAttribute(d));
                 }
-                if (!e && matches(s)) {
-                    if (t.tagName.toLowerCase() === 'a') {
-                        c = 'link';
+            } else {
+                let elm = t.tagName.toLowerCase();
+                if (elm !== 'html' && elm !== 'body') {
+                    if (t.id) {
+                        elm += `.${t.id}`;
```

Comment:
[suggestion]
I think `#` is more precise prefix than `.` for id because it is used for id on CSS.

Reply:
thanks @komlow.
applied your suggestion and added `className` as well.